src/event_listener.rs

```diff
@@ -0,0 +1,194 @@
+// Copyright 2025 Restate Software, Inc., Restate GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use std::ffi::CStr;
+use std::sync::Arc;
+
+use libc::c_void;
+
+use crate::ffi;
+use crate::ffi::{rocksdb_flushjobinfo_t, rocksdb_t};
+use crate::ffi_util::from_cstr;
+
+pub trait EventListenerExt {
+    fn add_event_listener<T>(&mut self, listener: Arc<T>)
+    where
+        T: EventListener + 'static + Send + Sync;
+}
+
+impl EventListenerExt for crate::Options {
+    fn add_event_listener<T>(&mut self, listener: Arc<T>)
+    where
+        T: EventListener + 'static + Send + Sync,
+    {
+        unsafe {
+            let cb = Box::new(EventListenerCallback::new(listener.clone()));
```

Comment:
```suggestion
            let cb = Box::new(EventListenerCallback::new(listener));
```
Unnecessary clone, if the Arc is actually needed.